src/qinterface/protected.cpp

```diff
@@ -71,9 +71,9 @@ void _expLog2x2(complex* matrix2x2, complex* outMatrix2x2, bool isExp)
 
     // Diagonal matrices are a special case.
     bool isDiag = true;
-    if (norm(matrix2x2[1]) > min_norm) {
+    if (norm(matrix2x2[1]) != 0) {
```

Comment:
No epsilon challenges with this?

Reply:
Performance would be hurt, and it is not, so, no. QFusion might benefit from a small nonzero epsilon, for optimization purposes, but it wouldn't typically make for a significant gain, for reasonably well-designed programs.

We're getting to place where I'm wondering if floating point error can be better constrained, and these changes go toward that end. Our informal and loose design philosophy has been that a small amount of unavoidable error can be quantified by testing and fed back into the library as a probability difference floor, which is `min_norm`. We need a more systematic approach, though, and ProjectQ complains, even if we're right to suppose that amplitude _consistency_ to like the 20th decimal place in unit tests, (that's just a made up number,) is moot when round-trip H gate error is much larger than that.